spyder/plugins/editor/widgets/base.py

```diff
@@ -204,23 +209,63 @@ def keyPressEvent(self, event):
             self.hide()
             QListWidget.keyPressEvent(self, event)
 
+    def up_to_date(self):
+        """
+        Checks if the selection is up to date
+        """
+        selected = self.currentItem().text()
+        completion_text = to_text_string(
+            self.textedit.get_current_word(completion=True))
+        if completion_text == 'None' and (
+                self.position == self.textedit.textCursor().position()):
+            return True
+        return selected.startswith(completion_text)
+
     def update_current(self):
+        """
+        Updates the completion_list and selects the best choice

```

Comment:
@impact27 the changes in this method are needed to prevent the error you fix with this PR? I think it is a nice improvement besides avoiding the error but just to be sure, since with this the way the completions list is populated changes (it reloads the list only with the elements that actually do a match):

Old behaviour:

![old](https://user-images.githubusercontent.com/16781833/57180540-1c6eb580-6e4f-11e9-9b9c-7801b250d721.gif)

New behaviour:

![new](https://user-images.githubusercontent.com/16781833/57180544-22fd2d00-6e4f-11e9-861e-c8cbf0a87463.gif)


what do you think @ccordoba12?

Reply:
On a separate note this should probably do fuzzy matching. See vscode example

<img width="637" alt="Screen Shot 2019-05-05 at 12 14 19" src="https://user-images.githubusercontent.com/3627835/57197633-5adba180-6f2f-11e9-857e-cfa3c9e91df2.png">

Reply:
> what do you think @ccordoba12?

I think it's fine. Thanks for noticing it @dalthviz!

Reply:
> On a separate note this should probably do fuzzy matching. See vscode example

It definitely should, but that may or may not be in scope for this PR.

Reply:
It is, @CAM-Gerlach could you open an issue for this :-p ?

Reply:
Well, I don't think LSP does that, so it would be out of scope. I mean I am only working with what I get from LSP here.

Reply:
@goanpeca Already exists (more or less) as #2374 (the other ask there was filtering the list to only matching items, which this PR already does), so I transcluded your comment over there.